### PR TITLE
BARON run file added

### PR DESCRIPTION
**Pull request by sanfranciscocarrot**
Should be able to work with BARON.jl. Currently, BARON.jl doesn't have MPB.getsolvetime() implemented so there can be a little bit issue with results grep. I also made a pull request to BARON.jl. Once that is merged, this BARON run should be smooth.

### CODE REVIEW

File: baron.jl [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env julia
+using JuMP
+using ArgParse
+
+using MINLPLib_jump
```
Comment (ccoffrin) [inline baron.jl]: Using this as a package is inconsistent with the other scripts which use the "include" method of importing the models from a file directly.  I have a slight preference for "include" approach.

File: baron.jl [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+using BARON
+
+
+include("common.jl")
+
+function parse_commandline()
```
Comment (ccoffrin) [inline baron.jl]: probably better to call this parse_commandline_baron

File: baron.jl [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+end
+
+
+function main(parsed_args)
+
+    m = fetch_model(parsed_args["file"])
+    # this adds a model named m to the current scope
+
+    # By default, always run with CPLEX
+    parsed_args["cpxlib"] != nothing ? cpxlib = parsed_args["cpxlib"] : cpxlib = "libcplex1271.so"
```
Comment (ccoffrin) [inline baron.jl]: Does this still work if cplex is not installed?

**Review by ccoffrin — COMMENTED** [review]
some minor suggestions

**Comment by ccoffrin** [comment]
regarding `getsolvetime()` this is the work around I have been using in PowerModels, https://github.com/lanl-ansi/PowerModels.jl/blob/a39d388957d59c899299a1f2dd388d6db7627b50/src/core/base.jl#L118, maybe we can use the same one here?

File: baron.jl (hunk 3 above)
Comment (sanfranciscocarrot) [reply baron.jl]: Yes. Baron will just say I cannot find the lib and switch back to Cbc.

**Comment by sanfranciscocarrot** [comment]
I like that solution with getsolvetime(). I think Joey just merge the latest pull request in BARON.jl, which means it should work now.
Also, suggested changes have been made.

**Comment by ccoffrin** [comment]
Looks good thanks!

**Pull request merged**